### PR TITLE
[CBRD-24419] Beautify and reorganize the query specification of the boot_define_view_ function.

### PR DESCRIPTION
**Pull request by youngjinj**
http://jira.cubrid.org/browse/CBRD-24419

### Beautify and reorganize the query specification of the boot_define_view_ function.
- Fixed the problem that the db_vclass result was incorrect
- If the data types do not match, the CAST function is used.
  ```
  <db_user>.[name]
  -> CAST (<db_user>.[name] AS VARCHAR(255)). /* string -> varchar(255) */

  <db_trigger>.[target_attribute]
  -> CAST (<db_trigger>.[target_attribute] AS VARCHAR (255)) /* string -> varchar(255) */
  ```
- Reorganize the query spec for db_partition.
  - Not using db_direct_super_class in FROM.
- Keep aliases consistent.
- Beautify the query specification of boot_define_view_ functions.

### CODE REVIEW

**Review by ctshim — APPROVED** [review]

File: src/transaction/boot_cl.c [hunk 1]
```diff
@@ -5456,81 +6035,57 @@ boot_define_view_partition (void)
...
-	  "CASE "
-	      "WHEN [p].[ptype] = 0 THEN 'HASH' "
-	      "WHEN [p].[ptype] = 1 THEN 'RANGE' "
-	      "ELSE 'LIST' "
-	      "END AS [partition_type], "
-	   "TRIM (SUBSTRING ([pi].[pexpr] FROM 8 FOR (POSITION (' FROM ' IN [pi].[pexpr]) - 8))) AS [partition_expr], "
-	   "[p].[pvalues] AS [partition_values], "
-	   "[p].[comment] AS [comment] "
+	  "CONCAT ([s].[class_name], '__p__', [p].[pname]) AS [partition_class_name], " /* It can exceed varchar(255). */
+	  "CASE [p].[ptype] WHEN 0 THEN 'HASH' WHEN 1 THEN 'RANGE' ELSE 'LIST' END AS [partition_type], "
```
Comment (beyondykk9) [inline src/transaction/boot_cl.c:6042]: it looks need to re-arange for CASE caluse.

File: src/transaction/boot_cl.c [hunk 2]
```diff
@@ -5735,14 +6321,34 @@ boot_define_view_db_collation (void)
...
-	   " WHEN 0 THEN 'Not applicable' WHEN 1 THEN 'Primary' WHEN 2 THEN 'Secondary'"
-	   " WHEN 3 THEN 'Tertiary' WHEN 4 THEN 'Quaternary' WHEN 5 THEN 'Identity' ELSE 'Unknown' END"
-	   " FROM [%s] [c] JOIN [%s] [ch] ON [c].[charset_id] = [ch].[charset_id] ORDER BY [c].[coll_id]",
-	   CT_COLLATION_NAME, CT_CHARSET_NAME);
+	"SELECT "
+	  "[coll].[coll_id] AS [coll_id], "
+	  "[coll].[coll_name] AS [coll_name], "
+	  "[ch].[charset_name] AS [charset_name], "
+	  "CASE [coll].[built_in] WHEN 0 THEN 'No' WHEN 1 THEN 'Yes' ELSE 'ERROR' END AS [is_builtin], "
+	  "CASE [coll].[expansions] WHEN 0 THEN 'No' WHEN 1 THEN 'Yes' ELSE 'ERROR' END AS [has_expansions], "
```
Comment (beyondykk9) [inline src/transaction/boot_cl.c:6331]: it looks need to re-arange for CASE caluses.

**Review by beyondykk9 — APPROVED** [review]

File: src/transaction/boot_cl.c (hunk 1 above)
Comment (youngjinj) [reply src/transaction/boot_cl.c:6042]: If there are not many branching values and does not exceed 120 bytes, it is written in one line.
If there is no other opinion, I will keep it.

File: src/transaction/boot_cl.c (hunk 2 above)
Comment (youngjinj) [reply src/transaction/boot_cl.c:6331]: If there are not many branching values and does not exceed 120 bytes, it is written in one line.
If there is no other opinion, I will keep it.

**Review by shparkcubrid — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

**Review by hyunikn — APPROVED** [review]

**Pull request merged**